### PR TITLE
Specify pool name when using REUSABLE client strategy

### PR DESCRIPTION
**Pull request by skoranda**
Add code to choose a random name to be used as the pool name when using
the REUSABLE client strategy. Without this enhancement two instances of
the microservice connecting to distinct servers will share a single
connection pool.

### All Submissions:

* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Have you added an explanation of what problem you are trying to solve with this PR?
* [X] Have you added information on what your changes do and why you chose this as your solution?
* [ ] Have you written new tests for your changes?

No new tests because the pool name is only used for real connections to real LDAP servers, not during mock ups.

* [X] Does your submission pass tests?
* [X] This project follows PEP8 style guide. Have you run your code against the 'flake8' linter?

### CODE REVIEW

File: src/satosa/micro_services/ldap_attribute_store.py [hunk 1]
```diff
@@ -303,6 +305,8 @@ def _ldap_connection_factory(self, config):
 
         pool_size = config["pool_size"]
         pool_keepalive = config["pool_keepalive"]
+        pool_name = ''.join(random.sample(string.ascii_lowercase, 6))
```
Comment (c00kiemon5ter) [inline src/satosa/micro_services/ldap_attribute_store.py]: I don't remember the entropy that `random` has; another idea may be to use `uuid.uuid4()`.

**Comment by c00kiemon5ter** [comment]
Is this ready for merge?

File: src/satosa/micro_services/ldap_attribute_store.py (hunk 1 above)
Comment (skoranda) [reply src/satosa/micro_services/ldap_attribute_store.py]: There are no crytographic considerations here. The random number is only used to ensure that two pools have different names. So the only requirement is that success calls to random() deliver N distinct numbers where N is the number of LDAP servers being contacted. It seems likely that N will be < 10 and random() is easily able to meet the requirement.

**Comment by skoranda** [comment]
Yes, ready for merge. Thanks.

**Pull request merged**